gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManagementTaskStreamImpl.java

```diff
@@ -146,18 +141,14 @@ public DagTask next() {
   private MultiActiveLeaseArbiter.LeaseAttemptStatus retrieveLeaseStatus(DagActionStore.DagAction dagAction)
       throws IOException, SchedulerException {
     MultiActiveLeaseArbiter.LeaseAttemptStatus leaseAttemptStatus;
-    if (!this.isMultiActiveExecutionEnabled) {
-      leaseAttemptStatus = new MultiActiveLeaseArbiter.LeaseObtainedStatus(dagAction, System.currentTimeMillis(), Long.MAX_VALUE, null);
-    } else {
-      // TODO: need to handle reminder events and flag them
-      leaseAttemptStatus = this.dagActionExecutionLeaseArbiter
-          .tryAcquireLease(dagAction, System.currentTimeMillis(), false, false);
-          /* Schedule a reminder for the event unless the lease has been completed to safeguard against the case where even
-          we, when we might become the lease owner still fail to complete processing
-          */
-      if (!(leaseAttemptStatus instanceof MultiActiveLeaseArbiter.NoLongerLeasingStatus)) {
-        scheduleReminderForEvent(leaseAttemptStatus);
-      }
+    // TODO: need to handle reminder events and flag them
+    leaseAttemptStatus = this.dagActionExecutionLeaseArbiter
+        .tryAcquireLease(dagAction, System.currentTimeMillis(), false, false);
```

Comment:
`DMTSImpl` could itself take in a `Supplier<Long> currentTimeMillisClock`

Reply:
Where would it get the supplier from? This is the only implementation and it's initialized by Guice. This code should be the one deciding how currentTimeMillis is determined.

Reply:
sorry, same as in the other comment:

> pair that with a ctor form taking such a param (for testing). the ctor w/o that extra param (for not testing) would simply pass in `System::currentTimeMillis`

basically, use the N-1 params ctor from Guice and the N params ctor for tests that need a predictable clock

Reply:
I don't think this is a parameter worth passing at a class level. If I have it configurable in the method then the caller (test class or non-test) can easily specify the value.